jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -170,36 +184,43 @@ public static boolean isSameName(Path pathA, Path pathB)
     @Override
     public boolean exists()
     {
-        return Files.exists(targetPath != null ? targetPath : path);
-    }
-
-    @Override
-    public boolean equals(Object obj)
-    {
-        if (this == obj)
-        {
-            return true;
-        }
-        if (obj == null)
+        if (aliasResolved)
         {
-            return false;
+            if (targetPath == null)
+                return false;
+            return Files.exists(targetPath);
         }
-        if (getClass() != obj.getClass())
+        else
         {
-            return false;
+            // Avoid resolving alias unless necessary to determine existence.
+            if (Files.exists(path))
+                return true;
+            if (isAlias() && targetPath != null)
+                return Files.exists(targetPath);
         }
-        PathResource other = (PathResource)obj;
-        return Objects.equals(path, other.path);
+
+        return false;
```

Comment:
don't need the else here as the previous block always returns.

Reply:
If the alias hasn't been resolved, and the original path (or target path) does not exist, then this is the final return.
This entire if/else block can be rewritten better.
Esp in light of the `Boolean` change.